### PR TITLE
Rebuild for CUDA 12 w/arch + Windows support

### PR DESCRIPTION
**Pull request by RaulPPelaez**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This is #119 reopened without merge conflicts.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,7 +44,7 @@ requirements:
     - swig                                   # [build_platform != target_platform]
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}  # [linux64 or ppc64le or win]
+    - {{ compiler('cuda') }}  # [linux or win]
```
Comment (jakirkham) [inline recipe/meta.yaml:47]: To ensure the CUDA compiler is only added when used, would suggest this tweak to the selector

```suggestion
    - {{ compiler('cuda') }}  # [(linux or win) and cuda_compiler != "None"]
```

**Review by jakirkham — COMMENTED** [review]
Thanks Raul! 🙏

Had a minor suggestion below to help with a couple CI failures that cropped up

**Review by mikemhenry — APPROVED** [review]

**Comment by mikemhenry** [comment]
LGTM

**Review by mikemhenry — CHANGES_REQUESTED** [review]
I changed my mind, the build number needs to be bumped

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Review by mikemhenry — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Mike! 🙏

Looks like it passes 🎉

**Comment by mikemhenry** [comment]
Forgot to add the automerge label :upside_down_face:

**Pull request merged**